constellations/gemini-admin/src/emails/accountInfo.js

```diff
@@ -0,0 +1,134 @@
+"use strict";
+require("../account/accountInfo.sass");
+import React from "react";
+import { Link } from "react-router-dom";
+import { getFullName } from "../utils/userUtils.js";
+
+export class AccountInfo extends React.Component {
+    state = {
+        id: 0,
+        email: "",
+        users: [],
+        transactions: [],
+        selectedUserEmails: [],
+        active: false,
+    };
+
+    formatCurrency = (amount) => {
+        return new Intl.NumberFormat("en-US", {
+            style: "currency",
+            currency: "USD",
+        }).format(amount);
+    };
+
+    onCheckUser = (e, userId) => {
+        const users = this.props.users;
+        if (e.target.checked) {
+            users.map((user, index) => {
+                if (user.id == userId) {
+                    var email = this.state.selectedUserEmails;
+                    email.push(user.email);
+                    this.setState({
+                        selectedUserEmails: email,
+                    });
+                }
+            });
+        } else {
+            users.map((user, index) => {
+                if (user.id == userId) {
+                    var email = this.state.selectedUserEmails;
+                    email.splice(email.indexOf(user.email), 1);
+                    this.setState({
+                        selectedUserEmails: email,
+                    });
+                }
+            });
+        }
+    };
+
+    render = () => {
+        const id = this.props.id;
+        const email = this.props.email;
+        const users = this.props.users;
+        const transactions = this.props.transactions;
+        const name = this.props.name;
+
+        var emails = [];
+        users.map((user, index) => {
+            emails.push(user.email);
+        });
+
+        const userRows = users.map((user, index) => {
+            var status = user.isGuardian ? "(guardian" : "(student";
+            status += user.email == email ? ", primary contact)" : ")";
+
+            if (this.state.selectedUserEmails.length > 0) {
+                this.state.selectedUserEmails.map((existingEmail, index) => {
+                    if (emails.indexOf(existingEmail) == -1) {
+                        this.setState({
+                            selectedUserEmails: [],
+                        });
+                    }
+                });
```

Comment:
filter or find here. Or maybe use forEach.

map() is used to create a list from another list.